lib/classes/game.rb

```diff
@@ -12,59 +12,66 @@ def switch_players
 			@current_player, @other_player = @other_player, @current_player
 		end
 
-		def solicit_move
-			"#{@current_player.name}: Enter a number between 1 and 9."
+		def solicit_move(again = false)
+			unless again
+				puts "#{@current_player.name}: Enter a number between 1 and 9.\n"
+			else
+				puts "#{@current_player.name}: Enter a number between 1 and 9. Please enter a number for an empty cell."
+			end
 		end
 
-		def get_move(move = gets.chomp)
-			keyboard_to_position(move)
-		end
+    def get_move(move = gets.chomp)
+      keyboard_to_position(move)
+    end
 
-		def game_over_mssg
-			return "#{@current_player.name} won!" if board.game_over == :winner
-			return 'The game ended in a tie' if board.game_over == :draw
-		end
+    def game_over_mssg
+      puts "#{@current_player.name} won!" if board.game_over == :winner
+      puts "The game ended in a tie" if board.game_over == :draw
+    end
 
-		def play
-			puts "#{@current_player.name} is the first player!"
-			while true
-				board.formatted_grid
-    		puts ''
-    		puts solicit_move
-				col, row = get_move
-				puts ''
-    		move = board.set_cell(row, col, @current_player.turn)
-    		while move.nil?
-    			puts solicit_move + ' Please enter a number in an empty cell.'
-					col, row = get_move
-					puts ''
-    			move = board.set_cell(row, col, @current_player.turn)
-    		end	
-				if @board.game_over
-					puts game_over_mssg
-					board.formatted_grid
-					return
-				else
-					switch_players
-				end
-			end
-		end
+    def print_board
+      board.formatted_grid
+    end
+
+    def play
+      puts "#{@current_player.name} is the first player!"
+      while true
+        print_board
+        solicit_move
+        col, row = get_move
+        move = board.set_cell(row, col, @current_player.avatar)
+        solicit_until(move)
```

Comment:
I'm curious. Why you use `solicit_until` here and before you use `print_board`, `solicit_move` and `col, row = get_move`?

Reply:
Because we only want to output a different message if they initially placed move in space that is already occupied, so we want the initial then if they move in empty space then we let them know. 
However, we can simplify.

Reply:
Resolved in commit  [acf4f8bba97343d448e80321113fee525ea50369](https://github.com/adriaanbd/tic-tac-toe/pull/18/commits/acf4f8bba97343d448e80321113fee525ea50369)